build.py

```diff
@@ -113,20 +113,71 @@ def get_distribution_name():
     except ImportError:
         return distro.codename()
 
-def get_package_filename(p):
+def get_distribution_type(distro_id = None):
+    if distro_id is None:
+        distro_id = distro.id()
+    if distro_id == 'ol': # special handling for Oracle Linux
+        return 'rhel'
+    if distro_id in ['rhel', 'ubuntu', 'suse']:
+        return distro_id
+    distro_like = distro.like().split()
+    if len(distro_like) == 0:
+        return distro_id
+    return get_distribution_type(distro_like[0])
+
+# Does not account for derivatives that follow their own version scheme
+# (ex: RHEL7-like Amazon Linux 2)
+def get_distribution_version():
+    distro_type = get_distribution_type()
+    if distro_type == 'ubuntu':
+        return '{0}.{1}'.format(distro.major_version(), distro.minor_version())
+    elif distro_type == 'suse':
+        return distro.version()
+    distro_version = distro.major_version()
+    if distro_version in ['', None]:
+        return 'rolling'
+    else:
+        return distro_version
+
+def get_package_name(p):
+    v = get_versions()[p]
+    return 'irods-externals-{0}{1}-{2}'.format(p, v['version_string'], v['consortium_build_number'])
+    
+def get_package_version(p):
     v = get_versions()[p]
+    ver_base = '1.0-'
+    ver_pkgrev = v.get('package_revision', '0')
+    ver_pkgrev_suffix1 = ''
+    ver_pkgrev_suffix2 = ''
+    if get_package_type() == 'deb':
+        ver_pkgrev_suffix1 = '~'
+        ver_pkgrev_suffix2 = get_distribution_name()
+    else:
+        dt = get_distribution_type()
+        dv = get_distribution_version()
+        ver_pkgrev_suffix2 = dv
+        if dt == 'rhel':
+            ver_pkgrev_suffix1 = '.el'
+        elif dt == 'fedora':
+            ver_pkgrev_suffix1 = '.fc'
+        else:
+            # probably suse
+            ver_pkgrev_suffix1 = '.{0}'.format(dt)
```

Comment:
@trel Is anyone running iRODS on suse (today)?

We don't publish packages for this AFAIK.

Reply:
We haven't published packages for suse since before I started working on irods

Reply:
Ok. I'm sure there are references to suse due to historical reasons. And its presence doesn't hurt anything.

Letting it ride.